### ISSUE TITLE
Optionally keeping previous Component non-destroyed while routing to a new Component

### ISSUE THREAD

**Issue by Bayramooov** [angular/angular#55207]
### Which @angular/* package(s) are relevant/related to the feature request?

core, router

### Description

Optionally, I want to keep my previous page as active when I want to open other pages. So, when I close this new page, I could go back to my previous page with everything left as it was.

Yeah, I know that I can save somewhere previous component's state, and re-init with that state data when I come back to that page, but it's not a good solution, here's why:

1. I cannot always save each state of a components, there might be some pure template (html/css/js, css animations, etc.) related things which I cannot save and re-init with.

2. Even if I can save each state of a component, it would be huge overhead to always check every time whether I have added this some new state into my state saving mechanism.

3. There might be really heavy to render components with lot's of initial calculation, so when I needed to switch to a different route for a second (maybe, select something from my list in a different route) and come back, I don't want it to re-render everything.
Instead, I want to keep my prevoius page as it is (like as if I just hide it) when I want to open some different, and want to come back in a second. And, maybe come back with some data even (but this is another feature..)

Some real-world scenarios:

While I am filling a hude Form, I need to fill select input, when I couldn't find proper item from a select input dropdown, I would want to add this item into that entity which should be done in a totally different route, so if I could open that route as a dialog* (page dialog - this is a name of my requesting feature), I wouldn't need to save everything on this page into some service, I would just add my new item and continue filling my form in my previous page.

### Proposed solution

As a solution, I think, I need some directive to use with routeLink, like: routerLinkAsDialog maybe. So, when I set this to true, I want Angular not to destroy my current component, and open new route component on the top of my current component. When I close that new page (with selecting something, or by saving something) I want to come to my last active page.

It is going to look like having layers of pages which are alive, so, when I close one, the last one going to open, etc.

### Alternatives considered

Alternattive considerations would be the ability to access data of my active pages. However, it is not that critical feature to have.

**Comment by eneajaho** [comment]
What you need is: `RouteReuseStrategy` that Angular already exposes. https://angular.dev/api/router/RouteReuseStrategy

https://tacettinsertkaya.medium.com/angular-routereusestrategy-335686fa7ad8

**Comment by atscott** [comment]
Yes, from what I understand, you're looking for `RouteReuseStrategy`. Closing in favor of #55216

**Comment by Bayramooov** [comment]
> What you need is: `RouteReuseStrategy` that Angular already exposes. https://angular.dev/api/router/RouteReuseStrategy
> 
> https://tacettinsertkaya.medium.com/angular-routereusestrategy-335686fa7ad8

Thank you, I didn't know about this feature, but I have one more question:

Can I set my route component as "Reusable" depending on each new route which I switch?
As I understand I should set some of route components as "Reusable". But that's not what I exactly expect:

In my case I want my components to be "reusable" depending on which route I'm switching:
let's assume I have 3 routes:
 /A, /B, /C for short.
 I want to set my routing mechanism as:
 if I route from /A into /B  => I want to reuse component of /A
 else if I route from /A into /C => I want it to work as standard (without reusing).
 
 @atscott , @eneajaho
